streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -59,6 +59,9 @@ private InternalAdminClientConfig(final Map<?, ?> props) {
 
     private final int retries;
     private final long retryBackOffMs;
+    private int remainingRetries;
+
+    private HashSet<String> leaderNotAvailableTopics = new HashSet<>();
```

Comment:
Oh, Nice catch! You're correct, @abbccdda , it might have a non-empty leaderNotAvailableTopics after each call to `makeReady`. I changed it to using local variable. Thank you.